x-pack/plugins/security_solution/public/entity_analytics/components/risk_information/index.tsx

```diff
@@ -52,18 +61,37 @@ const getTableColumns = (riskEntity?: RiskScoreEntity): Array<EuiBasicTableColum
   },
 ];
 
-interface TableItem {
-  range?: string;
-  level: RiskSeverity;
-}
-
-const tableItems: TableItem[] = [
+const riskLevelTableItems: RiskLevelTableItem[] = [
   { level: RiskSeverity.critical, range: i18n.CRITICAL_RISK_DESCRIPTION },
   { level: RiskSeverity.high, range: '70 - 90 ' },
   { level: RiskSeverity.moderate, range: '40 - 70' },
   { level: RiskSeverity.low, range: '20 - 40' },
   { level: RiskSeverity.unknown, range: i18n.UNKNOWN_RISK_DESCRIPTION },
 ];
+
+interface CriticalityLevelTableItem {
+  level: CriticalityLevel;
+  weight: number;
+}
+
+const criticalityLevelTableItems: CriticalityLevelTableItem[] = (
+  Object.keys(CriticalityModifiers) as CriticalityLevel[]
+).map((level) => ({ level, weight: CriticalityModifiers[level] }));
```

Comment:
in fact, I think we can use the `fp-ts/Record::mapWithIndex` function instead.
Although for proper type inference to work it might require piping to the `CriticalityModifiers` since I believe that fn is curried

Reply:
@tiansivive `keys` is _almost_ a type-correct drop-in replacement for `Object.keys`, except that it yields keys in alphabetical order (where `Object.keys` is definition order, and desired). I opted not to try to manually sort the list afterward, but that's a possibility.

`mapWithIndex` gives you an object, so not a full solution there; `collect` has the signature we want, but it also yields keys in alphabetical order.

I didn't change this for now, but happy to discuss this further and figure out a robust solution to this general issue as I've hit it a few times before and it doesn't seem like a priority for typescript (to be fair, their justification about runtime properties makes sense).

Reply:
Sorry for the delayed response, I was somehow unsubscribed

I'm ok with leaving as is, and definitely a +1 on figuring out a solution. I have noticed we use type assertions a lot in kibana, and that scary 😅 

In this case, if I'm understanding correctly, we want to preserve the keys order because we're using it for display purposes? 
I guess it's fine but I'd personally shy away from relying on key definition order.